xml/smb_native_os.xml

```diff
@@ -384,6 +384,33 @@
       <param pos="2" name="os.build"/>
    </fingerprint>
 
+      <!-- Mac OS X -->
+   <fingerprint pattern="^Samba (3\.0\.28a-apple)$">
```

Comment:
The `smbd --version` point is proof enough.  Lets keep it as is.

To the point about authenticated version checks, keep in mind that recog is a generic framework that is not specific to Nexpose, metasploit or even necessarily any Rapid7 product (although today they are the primary consumers).  We need to ensure that any changes we make here are not overloaded with things that are product specific.